src/main/java/budgetflow/command/AddIncomeCommand.java

```diff
@@ -0,0 +1,124 @@
+package budgetflow.command;
+
+import budgetflow.exception.InvalidNumberFormatException;
+import budgetflow.exception.MissingDateException;
+import budgetflow.exception.MissingAmountException;
+import budgetflow.exception.MissingCategoryException;
+import budgetflow.exception.MissingIncomeException;
+import budgetflow.expense.ExpenseList;
+import budgetflow.income.Income;
+import java.util.List;
+import java.util.logging.Logger;
+import budgetflow.parser.DateValidator;
+
+//@@author thienkimtranhoang
+public class AddIncomeCommand extends Command {
+    public static final String ERROR_INVALID_DATE = "Error: Date is not a valid date";
+    
+    private static final Logger logger = Logger.getLogger(AddIncomeCommand.class.getName());
+    private static final String ADD_COMMAND_PREFIX = "add ";
+    private static final int ADD_COMMAND_PREFIX_LENGTH = ADD_COMMAND_PREFIX.length();
+
+    private static final String ERROR_MISSING_INCOME_CATEGORY = "Error: Income category is required.";
+    private static final String ERROR_MISSING_INCOME_AMOUNT = "Error: Income amount is required.";
+    private static final String ERROR_MISSING_INCOME_DATE = "Error: Income date is required.";
+    private static final String ERROR_INCORRECT_INCOME_DATE = "Error: Income date is in wrong format." +
+            "please use DD-MM-YYYY format.";
+
+
```

Comment:
try to avoid illogical spacing so as to not trip-up the reader. Use blank lines to separate groups of statements, but i think 3 lines here is a lot. 1 line gap should be enough